kafka-client/src/main/java/dev/responsive/kafka/internal/stores/CommitBuffer.java

```diff
@@ -494,7 +501,11 @@ private void throwFencedException(final RemoteWriteResult result, final long con
         stored.offset
     );
     log.warn(msg);
-    throw new TaskMigratedException(logPrefix + msg);
+    if (eosEnabled) {
```

Comment:
I feel like it would be better to push this logic out as far to the periphery as possible and not couple the details of streams exception handling with the implementation of this class. What I mean is rather than storing `eosEnabled` as a field, we could just have something like `Supplier<RuntimeException> fencedExceptionFactory` and just call that here to get the exception.

I'm having trouble articulating why I like that better. It just feels off to have this detail of streams exception handling leaking all the way into this class.

Also, lets definitely leave a comment explaining why we need these exception types.

Reply:
That's fair -- moved the logic into a new ExceptionSupplier class, lmk if this makes more sense to you now

Tbh this might be useful for other kinds of exceptions as well, or at least somewhere to consolidate some of the error handling. For now it only supplies fenced exceptions but it could theoretically handle some of the other kinds of errors as our handling gets more sophisticated